test/python/visualization/test_circuit_text_drawer.py

```diff
@@ -319,21 +320,30 @@ def test_wire_order(self):
         """Test the wire_order option"""
         expected = "\n".join(
             [
-                "                    ",
-                "q_2: |0>────────────",
-                "        ┌───┐       ",
-                "q_1: |0>┤ X ├───────",
-                "        ├───┤ ┌───┐ ",
-                "q_3: |0>┤ H ├─┤ X ├─",
-                "        ├───┤ └─╥─┘ ",
-                "q_0: |0>┤ H ├───╫───",
-                "        └───┘┌──╨──┐",
-                " c: 0 4/═════╡ 0xa ╞",
-                "             └─────┘",
-                "ca: 0 2/════════════",
-                "                    ",
+                "                  ",
+                "q_2: |0>──────────",
+                "        ┌───┐     ",
+                "q_1: |0>┤ X ├─────",
+                "        ├───┤┌───┐",
+                "q_3: |0>┤ H ├┤ X ├",
+                "        ├───┤└─╥─┘",
+                "q_0: |0>┤ H ├──╫──",
+                "        └───┘  ║  ",
+                " c_2: 0 ═══════o══",
+                "               ║  ",
+                "ca_0: 0 ═══════╬══",
+                "               ║  ",
+                "ca_1: 0 ═══════╬══",
+                "               ║  ",
+                " c_1: 0 ═══════■══",
+                "               ║  ",
+                " c_0: 0 ═══════o══",
+                "               ║  ",
+                " c_3: 0 ═══════■══",
+                "              0xa ",
             ]
         )
+
```

Comment:
You're right. I think I had an early problem with wire_order and cregbundle off, but that went away. I'll change it back.

Reply:
Actually, I think there is a fundamental problem with wire_order changes to clbits and cregbundle on. How do you display the order if you have 3 qubits, and two 2-clbit registers with a wire order of [0, 2, 1, 6, 3, 4, 5]? I noticed that the tests for wire_order that are done in test_circuit_visualization only reorder the qubits. The reason this is an issue now is that with the control flow changes, the wire_map is used much more to access bit position than for example, self.qubits or self.clbits.

My suggestion would be to force cregbundle off if clbits are reordered.

Reply:
Actually (actually) I just realized this is that crazy problem with the fact that the text drawer tests unlike the other drawers bypass the call to circuit_drawer which is where all the checks to force cregbundle false are done.

The reality is that this is a bad test, since cregbundle would normally be forced off by circuit_drawer before the call to the text drawer. So I say, either ditch the test or do what I did here, which is explicitly make cregbundle false.

Reply:
So left this with cregbundle false which is the only way it should be tested. To avoid problems in the future, I think it's worthwhile to switch to using circuit_drawer for text_drawer testing.

Reply:
Yep, 100% agree.  Let's make that change in a separate PR, though, since it's a large-scale change to the testing structure.